Stop marking isInvalidInputType() as friend to InputType subclasses

cdumez opened the pull request:
#### 5b1d7c2431023876aa2cdc4035877580a48396ef
<pre>
Stop marking isInvalidInputType() as friend to InputType subclasses
<a href="https://bugs.webkit.org/show_bug.cgi?id=255731">https://bugs.webkit.org/show_bug.cgi?id=255731</a>

Reviewed by Darin Adler.

Stop marking isInvalidInputType() as friend to InputType subclasses. We just
need to make a few validation function public (which they are already in the
base class). This allows us to move isInvalidInputType() to InputType.cpp,
since this is the only place it is used.

* Source/WebCore/html/BaseDateAndTimeInputType.h:
* Source/WebCore/html/ButtonInputType.h:
* Source/WebCore/html/CheckboxInputType.h:
* Source/WebCore/html/ColorInputType.h:
* Source/WebCore/html/DateInputType.h:
* Source/WebCore/html/DateTimeLocalInputType.h:
* Source/WebCore/html/EmailInputType.h:
* Source/WebCore/html/FileInputType.h:
* Source/WebCore/html/HiddenInputType.h:
* Source/WebCore/html/ImageInputType.h:
* Source/WebCore/html/InputType.cpp:
(WebCore::isInvalidInputType):
(WebCore::InputType::typeMismatchFor const): Deleted.
(WebCore::InputType::typeMismatch const): Deleted.
(WebCore::InputType::valueMissing const): Deleted.
(WebCore::InputType::hasBadInput const): Deleted.
(WebCore::InputType::patternMismatch const): Deleted.
* Source/WebCore/html/InputType.h:
(WebCore::InputType::typeMismatchFor const):
(WebCore::InputType::valueMissing const):
(WebCore::InputType::hasBadInput const):
(WebCore::InputType::patternMismatch const):
(WebCore::InputType::typeMismatch const):
(WebCore::isInvalidInputType): Deleted.
* Source/WebCore/html/MonthInputType.h:
* Source/WebCore/html/NumberInputType.h:
* Source/WebCore/html/PasswordInputType.h:
* Source/WebCore/html/RadioInputType.h:
* Source/WebCore/html/RangeInputType.h:
* Source/WebCore/html/ResetInputType.h:
* Source/WebCore/html/SearchInputType.h:
* Source/WebCore/html/SubmitInputType.h:
* Source/WebCore/html/TelephoneInputType.h:
* Source/WebCore/html/TextInputType.h:
* Source/WebCore/html/TimeInputType.h:
* Source/WebCore/html/URLInputType.h:
* Source/WebCore/html/WeekInputType.h:

Canonical link: <a href="https://commits.webkit.org/263179@main">https://commits.webkit.org/263179@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6eaf9f97260bf094327f0423b6e0480af2339f63

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3871 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3961 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4070 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5306 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4130 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/3847 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4042 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3967 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/3777 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3919 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4115 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3491 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/5142 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1619 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3466 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/5016 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/3438 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/3526 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/4911 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3925 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/3186 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3447 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3466 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/940 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/3491 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3725 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6eaf9f97260bf094327f0423b6e0480af2339f63. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/12975

cdumez committed:
Stop marking isInvalidInputType() as friend to InputType subclasses
https://bugs.webkit.org/show_bug.cgi?id=255731

Reviewed by Darin Adler.

Stop marking isInvalidInputType() as friend to InputType subclasses. We just
need to make a few validation function public (which they are already in the
base class). This allows us to move isInvalidInputType() to InputType.cpp,
since this is the only place it is used.

* Source/WebCore/html/BaseDateAndTimeInputType.h:
* Source/WebCore/html/ButtonInputType.h:
* Source/WebCore/html/CheckboxInputType.h:
* Source/WebCore/html/ColorInputType.h:
* Source/WebCore/html/DateInputType.h:
* Source/WebCore/html/DateTimeLocalInputType.h:
* Source/WebCore/html/EmailInputType.h:
* Source/WebCore/html/FileInputType.h:
* Source/WebCore/html/HiddenInputType.h:
* Source/WebCore/html/ImageInputType.h:
* Source/WebCore/html/InputType.cpp:
(WebCore::isInvalidInputType):
(WebCore::InputType::typeMismatchFor const): Deleted.
(WebCore::InputType::typeMismatch const): Deleted.
(WebCore::InputType::valueMissing const): Deleted.
(WebCore::InputType::hasBadInput const): Deleted.
(WebCore::InputType::patternMismatch const): Deleted.
* Source/WebCore/html/InputType.h:
(WebCore::InputType::typeMismatchFor const):
(WebCore::InputType::valueMissing const):
(WebCore::InputType::hasBadInput const):
(WebCore::InputType::patternMismatch const):
(WebCore::InputType::typeMismatch const):
(WebCore::isInvalidInputType): Deleted.
* Source/WebCore/html/MonthInputType.h:
* Source/WebCore/html/NumberInputType.h:
* Source/WebCore/html/PasswordInputType.h:
* Source/WebCore/html/RadioInputType.h:
* Source/WebCore/html/RangeInputType.h:
* Source/WebCore/html/ResetInputType.h:
* Source/WebCore/html/SearchInputType.h:
* Source/WebCore/html/SubmitInputType.h:
* Source/WebCore/html/TelephoneInputType.h:
* Source/WebCore/html/TextInputType.h:
* Source/WebCore/html/TimeInputType.h:
* Source/WebCore/html/URLInputType.h:
* Source/WebCore/html/WeekInputType.h:

Canonical link: https://commits.webkit.org/263179@main

webkit-commit-queue commented:
Committed 263179@main (5b1d7c243102): <https://commits.webkit.org/263179@main>

Reviewed commits have been landed. Closing PR #12975 and removing active labels.
